src/emailVerification/index.tsx

```diff
@@ -0,0 +1,242 @@
+}: EmailVerificationProps) => {
+  /**
+   * 인증 번호에 대한 검사를 시작했는지 여부
+   */
+  const [checkAuthNumber, setCheckAuthNumber] = useState<boolean>(false);
+  /**
+   * 이메일 인증 횟수
+   */
+  const sendingEmailCount = useRef<number>(0);
+  const [authNumber, setAuthNumber] = useState<string | undefined>();
+  const verifiedEmail = useRef<string | undefined>();
+  /**
+   * 이메일 인증 시간 오버
+   */
+  const [openTimer, setOpenTimer] = useState<boolean>(false);
+  const [overTime, setOverTime] = useState<boolean>(false);
+  /**
+   * 이메일 인증 pass
+   */
+  const [pass, setPass] = useState<boolean>(false);
+  const [toastModalState, setToastModalState] = useState<ToastModalType>({
+    contents: '',
+    top: '0',
+    left: '0',
+  });
+  const nextBtnEl = document.querySelector('.next-btn');
+  const nextBtnElDomRect = nextBtnEl?.getClientRects()[0];
+  const successSendingEmail = useRef<boolean>(false);
+  /**
+   * 이메일 중복 여부, 서버에 인증 번호를 담은 이메일 요청등을 담당
+   */
+  const checkDuplicateEmail = () => {
+    let result: boolean = false;
+    //[api]이메일 중복 검사
+    return result;
+  };
+  const sendVerificationEmail = () => {
+    let result: boolean = true;
+    // [api] 인증 이메일 보내기
+    return result;
+  };
+  const onClickEmailBtn = async () => {
+    overTime && setOverTime(false);
+    const overSending: boolean = sendingEmailCount.current > 4;
+    // 1.가능한 이메일 인증 횟수를 충족한 경우
+    if (!overSending) {
+      //A. 백엔드에 이메일  중복 여부 확인
+      let duplicate: boolean = false; // 중복 이메일
+      if (isEmailDuplicate) {
+        const result = checkDuplicateEmail();
+        // result 값에 따라 duplicate 값 변경
+      }
+      // A-1 중복 메일인 경우
+      if (duplicate) {
+        setEmail((prev: InputDataType) => {
+          const newState: InputDataType = {
+            ...prev,
+            errorMsg: '이미 회원가입된 이메일이에요.',
+          };
+          return newState;
+        });
+      }
+      // A-2 유효한 메일
+      if (!duplicate) {
+        // a 서버에 이메일 인증 보내기
+        const result = sendVerificationEmail();
+        // 인증 번호 이메일 전송 성공
+        //  a-1 이메일 발송 성공
+        if (result) {
+          successSendingEmail.current = true;
+          // ㄱ. 타이머 작동. 모달 오픈
+          setOpenTimer(true);
+          if (nextBtnElDomRect !== null && nextBtnElDomRect !== undefined) {
+            // modal의 height, width에 따라 top,left 값 변경
+            setOpenToastModal(true);
+            setToastModalState({
+              contents: '인증 이메일이 발송됐어요.',
+              //  39:toastModal.height
+              top: `${nextBtnElDomRect.top - 39 - 16}px`,
+              left: `20vw`,
+            });
+            sendingEmailCount.current += 1;
+          }
+        }
+      }
+    }
+    //  2. 할 수 있는 이메일 인증 횟수를 초과한 경우
+    if (overSending) {
+      setOpenToastModal(false);
+      setTimeout(() => {
+        setOpenAlertModal(true);
+      }, 100);
+    }
+  };
+  const onChangeAuthNumber = (event: ChangeEvent<HTMLInputElement>) => {
+    const text = XSSCheck(event.target.value, undefined);
```

Comment:
XSSCheck 타입 중 level이 `level: undefined | number`로 되어있던데 `level?: number`로 수정 후 다음과 같이 수정해도 좋을 것 같습니다.
```suggestion
    const text = XSSCheck(event.target.value);
```


그리고 XSSCheck가 login에 존재하는데 다른 곳에서도 사용할 수 있으니 util로 빼도 좋을 것 같습니다.
`src/utils/function.ts`를 만들어서 위치시키면 좋겠네요. 당장 떠오르는 이름이 function밖에 없는데 그 외에 다른 이름을 사용해도 좋을 것 같습니다.